Add ecosystem wallet security documentation (#4911)
thirdweb-dev/js: apps/portal/src/app/connect/ecosystems/security/page.mdx

import { DocImage, Callout, createMetadata } from "@doc";

export const metadata = createMetadata({
	image: {
		title: "Ecosystem Wallet Security",
		icon: "wallets",
	},
	title: "Ecosystem Wallet Security",
	description:
		"Learn how thirdweb protects your user's private keys across an ecosystem using nitro Enclaves",
});

# Ecosystem Wallet Security

## Wallet Creation

When a user signs into an application using their email or social logins for the first time, a wallet is generated within a secure enclave on the server after verifying the user's legitimacy. The enclave provides a trusted execution environment, ensuring the wallet creation process is isolated and protected from external interference.

- The wallet and its corresponding private key are generated entirely within the enclave, never leaving its secure confines.
- User authentication data is verified within the enclave, ensuring that only legitimate, authenticated users can initiate wallet creation.

## Security Measures

- The enclave's cryptographic properties ensure that even the server operators cannot access the contents or operations within the enclave.

> - The enclave provides a verifiable hash of the image of the code that is being run on the device, allowing anyone to verify the contents of the code for malicious intent.

Reviewer: ```suggestion
- The enclave provides a verifiable hash of the image of the code that is being run on the device, allowing anyone to verify the contents of the code.
```

- The wallet's private key never exists in an unencrypted form outside of the enclave.

> - All sensitive operations, such as transaction signing, occur within the enclave, further protecting the wallet's security.

Reviewer: ```suggestion
- All sensitive operations, such as transaction signing, occur within the enclave, ensuring the wallet's security.
```

## Wallet Usage

### General Usage

- When users interact with their wallet via the the enclave, all communications are encrypted and only the legitimate user can access their wallet.
- All traffic is encrypted with TLS and HSTS. Services are run in private VPCs on AWS and accessible only from a single entry point via our Cloudflare DNS.

> - Applications can [link multiple authentication methods](/link-multiple-profiles) to the same enclave. Any of these methods can be used to authenticate into the users wallet.

Reviewer: ```suggestion
- Applications can [link multiple authentication methods](/connect/in-app-wallet/guides/link-multiple-profiles) to the same enclave. Any of these methods can be used to authenticate into the users wallet.
```

### Ecosystem Usage

- Ecosystem wallets are controlled by the ecosystem owner.

>   - We replaced the SSS model tricky with Encalves because a malicious developer could reverse engineer and extract the private key from the client and exploit users who might’ve assets from other developers within the ecosystem.

Reviewer: ```suggestion
  - We moved away from the shamir secret sharing model and replaced it with Encalves. This ensures no-one can extract the private key from the client, including other developers in the ecosystem.
```

>   - With enclave and partner policies, this is mitigated. Ecosystem owners are able to place restrictions on what individual partners are able to do. Moreover, since the private key is never re-constructed on the client, the ecosystem partner must submit the request to the enclave which would be able to verify the request and block requests that are out of scope or denied by the ecosystem owner.

Reviewer: ```suggestion
  - With enclave and partner policies, ecosystem owners are able to place restrictions on what individual partners are able to do. Since the private key is never re-constructed on the client, the ecosystem partner must submit the request to the enclave which would be able to verify the request and block requests that are out of scope or denied by the ecosystem owner.
```

## Recoverability

- Users can export their private key at any time.
- Thirdweb wallets support three categories of authentication: socials, custom authentication, and email / phone authentication. If a user ever loses access to their authentication method:

  1.  **For socials and email / phone authentication,** users can utilize the recovery flow of their providers to regain access to their account.
  2.  **In the case of custom authentication,** the developer managing their authentication flow will be able to re-instate the users account upon successful verification.

>   3.  If application providers **do not have direct access** to user accounts or private keys, as these remain secured within the enclave. The enclave's design ensures that only verified user requests can trigger wallet operations.

Reviewer: ```suggestion
  3.  Application providers **do not have direct access** to user accounts or private keys, as these remain secured within the enclave. The enclave's design ensures that only verified user requests can trigger wallet operations.
```
